### PR TITLE
Fix bugs in cifs hourly simulation

### PR DESCRIPTION
**Pull request by yzhou601**
## Pull Request Description

The hourly simulation always stepped into conditions of opening cfis fan, where the fan runtime fraction most recently read is not applied correctly. This PR is aimed to fix this issue while keeping its approach capability of scaling into smaller time step simulations.

## Checklist

Not all may apply:

- [ ] EPvalidator.rb has been updated
- [ ] Tests (and test files) have been updated
- [ ] Documentation has been updated
- [ ] `rake update_measures` has been run
- [ ] No unexpected regression test changes on CI

### CODE REVIEW

**Comment by yzhou601** [comment]
@jonwinkler do you want to take look at code diffs to make sure I was correctly understanding your approach and making reasonable changes?

File: resources/airflow.rb [hunk 1]
```diff
@@ -1708,36 +1710,34 @@ def self.create_infil_mech_vent_objects(model, building, infil, mech_vent, wind_
...
+      infil_program.addLine("      Set #{mech_vent.cfis_f_damper_open_var.name} = cfis_fan_runtime/(ZoneTimeStep*60)")
       infil_program.addLine("      Set QWHV = #{mech_vent.cfis_f_damper_open_var.name}*CFIS_Q_duct")
       infil_program.addLine("      Set #{mech_vent.cfis_t_sum_open_var.name} = CFIS_t_min_hr_open")
       infil_program.addLine("    Else")
       # Damper is open and using call for heat/cool to supply fresh air
-      infil_program.addLine("      Set cfistemp6 = fan_rtf*ZoneTimeStep*60")
-      infil_program.addLine("      Set #{mech_vent.cfis_t_sum_open_var.name} = #{mech_vent.cfis_t_sum_open_var.name}+cfistemp6")
-      infil_program.addLine("      Set #{mech_vent.cfis_f_damper_open_var.name} = 1")
-      infil_program.addLine("      Set QWHV = fan_rtf*CFIS_Q_duct")
+      infil_program.addLine("      Set cfis_fan_runtime = fan_rtf_hvac*ZoneTimeStep*60")
```
Comment (yzhou601) [inline resources/airflow.rb]: I've changed some names of EMS variables to help myself better understand the program. If they're not appropriate or making EMS harder to read, please let me know.

**Comment by shorowit** [comment]
@yzhou601 If I'm reading the code correctly, it looks like the CFIS duct losses are also impacted?

Can you summarize what energy changes you expect to see relative to the previous implementation for fan power, airflow heat transfer, and duct losses?

**Comment by shorowit** [comment]
Tested simulations with 1-hour and 10-min timesteps and fan energy numbers look much better now. Just need to double-check duct losses.

File: resources/airflow.rb [hunk 2]
```diff
@@ -1490,6 +1490,7 @@ def self.create_cfis_objects(model, building, mech_vent)
     mech_vent.cfis_t_sum_open_var = OpenStudio::Model::EnergyManagementSystemGlobalVariable.new(model, "#{Constants.ObjectNameMechanicalVentilation.gsub(" ", "_")}_cfis_t_sum_open") # Sums the time during an hour the CFIS damper has been open
     mech_vent.cfis_on_for_hour_var = OpenStudio::Model::EnergyManagementSystemGlobalVariable.new(model, "#{Constants.ObjectNameMechanicalVentilation.gsub(" ", "_")}_cfis_on_for_hour") # Flag to open the CFIS damper for the remainder of the hour
     mech_vent.cfis_f_damper_open_var = OpenStudio::Model::EnergyManagementSystemGlobalVariable.new(model, "#{Constants.ObjectNameMechanicalVentilation.gsub(" ", "_")}_cfis_f_damper_open") # Fraction of timestep the CFIS damper is open. Used by infiltration and duct leakage programs
```
Comment (shorowit) [inline resources/airflow.rb]: It seems like `cfis_f_damper_open_var` no longer needs to be a GlobalVariable now that you're using `cfis_f_damper_extra_open_var` in the duct program instead, right? Can you check if the others are still used?

File: resources/airflow.rb [hunk 3]
```diff
@@ -1509,6 +1510,7 @@ def self.create_cfis_objects(model, building, mech_vent)
     cfis_program.addLine("Set #{mech_vent.cfis_t_sum_open_var.name} = 0")
     cfis_program.addLine("Set #{mech_vent.cfis_on_for_hour_var.name} = 0")
     cfis_program.addLine("Set #{mech_vent.cfis_f_damper_open_var.name} = 0")
```
Comment (shorowit) [inline resources/airflow.rb]: If any global variables are removed above, remove here too.

File: resources/airflow.rb [hunk 4]
```diff
@@ -1708,36 +1710,34 @@ def self.create_infil_mech_vent_objects(model, building, infil, mech_vent, wind_
...
-      infil_program.addLine("    Else")
-      infil_program.addLine("      Set cfistemp1 = (@Mod (60.0-CFIS_t_fan_on) (60.0*ZoneTimeStep))")
-      infil_program.addLine("      Set #{mech_vent.cfis_f_damper_open_var.name} = cfistemp1/(60.0*ZoneTimeStep)") # calculates the portion of the current timestep the CFIS damper needs to be open
-      infil_program.addLine("      Set #{mech_vent.cfis_on_for_hour_var.name} = 1") # CFIS damper will need to open for all the remaining timesteps in this hour
-      infil_program.addLine("    EndIf")
+      # Evaluate condition of if supply fan have to run to achieve target minutes per hour of operation
+      infil_program.addLine("  If (Minute+0.00001) >= CFIS_t_fan_on")
+      infil_program.addLine("    Set #{mech_vent.cfis_on_for_hour_var.name} = 1") # on flag for duct program
+      # Consider fan rtf read in current calling point (results of previous time step) + CFIS_t_fan_on based on min/hr requirement and previous EMS results.
+      infil_program.addLine("    Set cfis_fan_runtime1 = @Max (@ABS(Minute - CFIS_t_fan_on)) (fan_rtf_hvac * ZoneTimeStep * 60)")
```
Comment (shorowit) [inline resources/airflow.rb]: The use of `cfis_fan_runtime`, `cfis_fan_runtime1`, and `cfis_fan_runtime2` is a bit confusing. Is it possible to use better names or consolidate them?

Comment (yzhou601) [reply resources/airflow.rb]: Lol sorry for that, I was trying to keep the final cfis_fan_runtime easier to track in edd so I made the process variables whatever names they can be. Will it be better to overwrite them with the same name?

Comment (shorowit) [reply resources/airflow.rb]: That seems reasonable to me.

**Comment by shorowit** [comment]
Thanks @jonwinkler.

Note that base-mechvent-cfis.xml has a significantly oversized HVAC system in it, so the fan runtime fractions are always pretty low and the differences are not as pronounced. If you change the file to have autosized HVAC equipment, the difference increases from 14% to 25%.

**Comment by yzhou601** [comment]
Thanks for thorough tests done by @jonwinkler  and @shorowit !

**Comment by shorowit** [comment]
Made one minor change. This is good to go. Will merge when the CI comes back ✔️.
